### PR TITLE
Add GitHub ci

### PR DESCRIPTION
**Pull request by zekth**
Adding github ci instead of travis

Also for the deploy we need a gh token

Seems to be 30s faster than travis

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+    if: github.ref == 'refs/heads/master'
+    steps:
+      - name: Download Build
+        uses: actions/download-artifact@master
+        if: github.ref == 'refs/heads/master'
+        with:
+          name: build
+          path: dist
+      - name: GitHub Pages Deploy
+        uses: zekth/gh-deploy-action@1.0.0
```
Comment (yanfali) [inline .github/workflows/build.yml]: why does this depend on a personal repo?

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+        with:
+          dir: dist
+          gh_token: ${{ secrets.GH_TOKEN }}
```
Comment (yanfali) [inline .github/workflows/build.yml]: I don't have permissions to create one of these. Where do we get these from?

File: .github/workflows/build.yml (hunk 1 above)
Comment (zekth) [reply .github/workflows/build.yml]: There is no official repo for pushing to gh-pages.

File: .github/workflows/build.yml (hunk 2 above)
Comment (zvecr) [reply .github/workflows/build.yml]: In this case its provided for you <https://help.github.com/en/actions/automating-your-workflow-with-github-actions/authenticating-with-the-github_token>

Comment (zekth) [reply .github/workflows/build.yml]: You can create one for the qmk_bot in the security settings of it

Comment (zekth) [reply .github/workflows/build.yml]: Or like zvecr said you can also use the generated one. I think it will work also

Comment (zvecr) [reply .github/workflows/build.yml]: Yeah I'm using it with a different action and it seems fine on push master

Comment (zekth) [reply .github/workflows/build.yml]: I'll change it. Thanks @zvecr

File: .github/workflows/build.yml (hunk 1 above)
Comment (yanfali) [reply .github/workflows/build.yml]: That seems odd

**Comment by yanfali** [comment]
@zekth eli5 is this going to work with our existing configuration you think? I'm a little uncomfortable using a fork from your personal account, I think it would be better if qmk forked your repo and just had a pointer? especially for a deployment task to a public service, it's a little bit risky. What if you lost control of the repo?

**Comment by zekth** [comment]
As long as there is not official github page to actions we can fall back to the deploy.sh script. This needs a bit of tune up but will work also

**Comment by yanfali** [comment]
@zekth go ahead and make it work with deploy.sh and we can merge this if you think it'll work as is

File: .github/workflows/build.yml [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+          name: build
+          path: dist
+  deploy:
+    needs: ['build','test']
+    runs-on: ubuntu-latest
+    if: github.ref == 'refs/heads/master'
+    steps:
+      - name: Download Build
+        uses: actions/download-artifact@master
+        if: github.ref == 'refs/heads/master'
```
Comment (zvecr) [inline .github/workflows/build.yml]: the job.if covers this case so can be safely removed
```suggestion
```

Comment (zekth) [reply .github/workflows/build.yml]: Haha good catch

Comment (yanfali) [reply .github/workflows/build.yml]: let me know tomorrow when you are around and we can try merging this and see what breaks.

**Comment by zekth** [comment]
@zvecr can you review the changes?

**Comment by yanfali** [comment]
@zvecr if you get a chance
